packager/mpd/base/xml/xml_node.cc

```diff
@@ -432,9 +432,12 @@ bool RepresentationXmlNode::AddVODOnlyInfo(const MediaInfo& media_info) {
 
 bool RepresentationXmlNode::AddLiveOnlyInfo(
     const MediaInfo& media_info,
-    const std::list<SegmentInfo>& segment_infos,
-    uint32_t start_number) {
+    const std::list<SegmentInfo>& segment_infos) {
   XmlNode segment_template("SegmentTemplate");
+
+  int start_number =
+      segment_infos.empty() ? 1 : segment_infos.begin()->start_segment_number;
+
```

Comment:
This is only needed for line 459 below. I'd prefer just skipping the block at line 456 if segment_infos is empty.

Reply:
It is also needed at line 465 as IsTimelineConstantDuration() argument.
Do you prefer having int start_number = ... in IsTimelineConstantDuration() method?

Reply:
`I'd prefer just skipping the block at line 456 if segment_infos is empty.
`
@kqyang With the above change, how do you want to handle RepresentationClone unit test?

Reply:
I think we can add a call to AddSegments once after calling CopyRepresentation. Can you see what happens with that?

Reply:
@kqyang I just saw your reply. Sorry about the delay. AddSegments is specific to representation_ in SegmentTemplateTest. Do you want to create a new method AddingSegmentsToClone and a new data member cloned_representation?
@vish91 I am still looking into this. Will update the PR after that.

Reply:
I have not addressed this in the PR update.